### PR TITLE
Ticket 4573: SANS2D named aperture and guide positions

### PR DESCRIPTION
**Pull request by DominicOram**
See https://github.com/ISISComputingGroup/IBEX/issues/4573.

* Added tests that cover 10 axes of motion set points.
* Converted tests to use galils and test `.cmd` (more like how it is run on a real instrument)
* Removed non-axis tests
* Added tests for alarm/unit propogation

### CODE REVIEW

File: tests/motion_setpoints.py [hunk 1]
```diff
@@ -153,120 +164,138 @@ def test_GIVEN_2D_no_axis_WHEN_set_position_THEN_position_is_set_and_motor_moves
...
+        self._test_alarm_propogates(self.ca1D, 1)
+
+    def test_GIVEN_2D_WHEN_second_axis_in_alarm_THEN_position_in_alarm(self):
+        self._test_alarm_propogates(self.ca2D, 2)
+
+    def test_GIVEN_10D_WHEN_set_position_THEN_position_is_set_and_motor_moves_to_position(self):
+        def check_motor_positions(expected_coords, readback):
+            pv_suffix = ".RBV" if readback else ""
+            for index, coord in enumerate(expected_coords):
+                controller = 1 if index < 8 else 2
```
Comment (JamesKingWork) [inline tests/motion_setpoints.py:271]: Suggestion: Move 8 out to well named constant with comment pointing towards the use of 8 axes per galil ioc

File: tests/motion_setpoints.py [hunk 2]
```diff
@@ -153,120 +164,138 @@ def test_GIVEN_2D_no_axis_WHEN_set_position_THEN_position_is_set_and_motor_moves
...
+                axis = (index % 8) + 1
+                self.motor_ca.assert_that_pv_is_number("MTR0{}0{}{}".format(controller, axis, pv_suffix), coord, 0.1)
```
Comment (JamesKingWork) [inline tests/motion_setpoints.py:273]: Suggestion: use timeout=0.1 to enhance readability

File: tests/motion_setpoints.py [hunk 3]
```diff
@@ -153,120 +164,138 @@ def test_GIVEN_2D_no_axis_WHEN_set_position_THEN_position_is_set_and_motor_moves
...
+        self.ca2D.assert_that_pv_is("IPOSN", 1)
+        self.ca2D.assert_that_pv_is("POSITIONED", 1)
+
+    def _test_alarm_propogates(self, channel_access, motor_num):
+        assert_alarm_state_of_posn(channel_access, motor_num, ChannelAccess.Alarms.NONE)
+
+        self.motor_ca.set_pv_value("MTR010{}.HLSV".format(motor_num), "MAJOR")
+        current_posn = self.motor_ca.get_pv_value("MTR010{}".format(motor_num))
+        self.motor_ca.set_pv_value("MTR010{}.HLM".format(motor_num), current_posn - 1)
+        self.motor_ca.assert_that_pv_alarm_is("MTR010{}".format(motor_num), ChannelAccess.Alarms.MAJOR)
```
Comment (JamesKingWork) [inline tests/motion_setpoints.py:257]: Suggestion: have a comment describing why we need to set MAJOR and move the motor to being out of range

File: tests/motion_setpoints.py [hunk 4]
```diff
@@ -153,120 +164,138 @@ def test_GIVEN_2D_no_axis_WHEN_set_position_THEN_position_is_set_and_motor_moves
...
+
+        assert_alarm_state_of_posn(channel_access, motor_num, ChannelAccess.Alarms.MAJOR)
+
+    def test_GIVEN_1D_WHEN_axis_in_alarm_THEN_position_in_alarm(self):
+        self._test_alarm_propogates(self.ca1D, 1)
+
+    def test_GIVEN_2D_WHEN_second_axis_in_alarm_THEN_position_in_alarm(self):
+        self._test_alarm_propogates(self.ca2D, 2)
```
Comment (JamesKingWork) [inline tests/motion_setpoints.py:265]: Why do we not have a test for 10D as well?

File: tests/motion_setpoints.py [hunk 5]
```diff
@@ -121,30 +132,30 @@ def test_GIVEN_1D_WHEN_set_offset_THEN_in_position_light_goes_off_then_on_and_mo
...
         self.ca1D.assert_that_pv_is("POSITIONED", 1)
         self.ca1D.assert_that_pv_is("POSN", POSITION_OUT, timeout=30)
 
-    def test_GIVEN_1D_no_axis_WHEN_get_numaxes_THEN_return1(self):
+    def test_GIVEN_1D_WHEN_get_numaxes_THEN_return1(self):
         self.ca1D.assert_that_pv_is("NUMAXES", 1)
 
-    def test_GIVEN_2D_no_axis_WHEN_get_numaxes_THEN_return2(self):
+    def test_GIVEN_2D_WHEN_get_numaxes_THEN_return2(self):
         self.ca2D.assert_that_pv_is("NUMAXES", 2)
```
Comment (JamesKingWork) [inline tests/motion_setpoints.py:156]: Why do we not test this for 10D as well?

File: tests/motion_setpoints.py [hunk 6]
```diff
@@ -37,82 +44,86 @@
...
         for index, (expected_position, expected_motor_position) in enumerate(POSITIONS_1D):
 
             self.ca1D.set_pv_value("IPOSN:SP", index)
 
             self.ca1D.assert_that_pv_is("IPOSN", index)
             self.ca1D.assert_that_pv_is("POSN", expected_position)
-            self.motor_ca.assert_that_pv_is("MTR0.RBV", expected_motor_position)
+            self.motor_ca.assert_that_pv_is("MTR0101.RBV", expected_motor_position)
 
     def test_GIVEN_1D_WHEN_move_to_out_position_THEN_in_position_light_goes_off_then_on(self):
```
Comment (JamesKingWork) [inline tests/motion_setpoints.py:117]: Can we have a test for 10D as well

File: tests/motion_setpoints.py [hunk 7]
```diff
@@ -37,82 +44,86 @@
...
 
         self.ca1D.assert_that_pv_is("POSITIONED", 0)
-        self.ca1D.assert_that_pv_is("STATIONARY", 0)
+        self.ca1D.assert_that_pv_is("STATIONARY0", 0)
 
         self.ca1D.assert_that_pv_is("POSITIONED", 1)
-        self.ca1D.assert_that_pv_is("STATIONARY", 1)
+        self.ca1D.assert_that_pv_is("STATIONARY0", 1)
 
     def test_GIVEN_1D_WHEN_set_offset_THEN_in_position_light_goes_off_then_on_and_motor_moves_to_position_plus_offset(self):
```
Comment (JamesKingWork) [inline tests/motion_setpoints.py:128]: Can we have a test for 10D as well

File: tests/motion_setpoints.py [hunk 8]
```diff
@@ -121,30 +132,30 @@ def test_GIVEN_1D_WHEN_set_offset_THEN_in_position_light_goes_off_then_on_and_mo
...
+        self.ca1D.set_pv_value("COORD0:OFFSET:SP", expected_offset)
         self.ca1D.assert_that_pv_is("POSITIONED", 0)
 
         self.ca1D.assert_that_pv_is("POSITIONED", 1)
-        self.ca1D.assert_that_pv_is("COORD1:OFFSET", expected_offset)
-        self.motor_ca.assert_that_pv_is("MTR0.RBV", expected_motor_position)
+        self.ca1D.assert_that_pv_is("COORD0:OFFSET", expected_offset)
+        self.motor_ca.assert_that_pv_is("MTR0101.RBV", expected_motor_position)
 
     def test_GIVEN_1D_WHEN_set_large_offset_THEN_current_position_set_correctly(self):
```
Comment (JamesKingWork) [inline tests/motion_setpoints.py:142]: Can we have a test for 10D as well

File: tests/motion_setpoints.py [hunk 9]
```diff
@@ -153,120 +164,138 @@ def test_GIVEN_2D_no_axis_WHEN_set_position_THEN_position_is_set_and_motor_moves
...
-
     def test_GIVEN_file_WHEN_duplicate_name_value_THEN_load_no_positions(self):
         self.caDN.assert_that_pv_is("POSN:NUM", 0)
-        self.caDN.assert_that_pv_is("NUMAXES", 0)
 
     def test_GIVEN_file_WHEN_duplicate_position_value_THEN_load_no_positions(self):
         self.caDP.assert_that_pv_is("POSN:NUM", 0)
-        self.caDP.assert_that_pv_is("NUMAXES", 0)
 
     def test_GIVEN_2D_WHEN_invalid_position_specified_THEN_alarm(self):
```
Comment (JamesKingWork) [inline tests/motion_setpoints.py:223]: Can we have a test for 10D as well

File: tests/motion_setpoints.py [hunk 10]
```diff
@@ -153,120 +164,138 @@ def test_GIVEN_2D_no_axis_WHEN_set_position_THEN_position_is_set_and_motor_moves
...
-        self.ca2DAxis.set_pv_value("POSN:SP", "an_invalid_position")
-        self.ca2DAxis.assert_that_pv_alarm_is("POSN:SP", ChannelAccess.Alarms.INVALID)
-        
+        self.ca2D.set_pv_value("POSN:SP", "an_invalid_position")
+        self.ca2D.assert_that_pv_alarm_is("POSN:SP", ChannelAccess.Alarms.INVALID)
+
     def test_GIVEN_2D_WHEN_move_motor_THEN_tolerance_checked(self):
```
Comment (JamesKingWork) [inline tests/motion_setpoints.py:227]: Can we have a test for 10D as well

File: tests/motion_setpoints.py [hunk 11]
```diff
@@ -37,82 +44,86 @@
...
 
 
+def assert_alarm_state_of_posn(channel_access, coordinate, expected_state):
+    channel_access.assert_that_pv_alarm_is("FORWARD_ALARM", expected_state)
+    channel_access.assert_that_pv_alarm_is("POSN", expected_state)
+    channel_access.assert_that_pv_alarm_is("POSITIONED", expected_state)
+    channel_access.assert_that_pv_alarm_is("POSN_NO_ERR", ChannelAccess.Alarms.NONE)
+
+
 class MotionSetpointsTests(unittest.TestCase):
```
Comment (JamesKingWork) [inline tests/motion_setpoints.py:66]: Can we add a test that changes the definition file contents and then reloads the file by writing to RESET and check that the POSITIONS PV updates. Also, a test to see that if we change the contents to something invalid then the POSITIONS PV is empty or in error.

**Review by JamesKingWork — DISMISSED** [review]
Extra test request

**Review by JamesKingWork — APPROVED** [review]
Looks good
